src/blockly/generators/propc/gpio.js

```diff
@@ -1710,19 +1710,7 @@ Blockly.Blocks.sd_read = {
     init: function () {
         this.setTooltip(Blockly.MSG_SD_READ_TOOLTIP);
         this.setColour(colorPalette.getColor('output'));
-        this.appendValueInput("SIZE")
-                .setCheck(null)
-                .appendField("SD file")
-                .appendField(new Blockly.FieldDropdown([
-                    ["read", "fread"],
-                    ["write", "fwrite"],
-                    ["close", "fclose"]
-                ], function (mode) {
-                    this.sourceBlock_.setSdMode(mode);
-                }), "MODE");
-        this.appendDummyInput("VALUE")
-                .appendField("bytes  store in")
-                .appendField(new Blockly.FieldVariable(Blockly.LANG_VARIABLES_SET_ITEM), 'VAR');
+        this.setSdMode('fwrite');
```

Comment:
Since the `setSdMode()` function was building this part of the block anyway, just call it instead to remove redundant code.